darwin/cli_functions.py

```diff
@@ -546,7 +546,7 @@ def dataset_convert(dataset_slug, format, output_dir):
         annotations_path = release_path / "annotations"
         if output_dir is None:
             output_dir = release_path / "other_formats" / f"{format}"
-            output_dir.mkdir(parents=True, exist_ok=True)
+        output_dir.mkdir(parents=True, exist_ok=True)
```

Comment:
I think we have a bug here. We added the type annotation for `output_dir`, but that doesn't guarantee that the argument is actually of that type.

Make sure that each time we call `dataset_convert` we actually make `output_dir` a `Path` (I don't think we're doing that, but I might be wrong)